agent/utils/ephemeral_ports.go

```diff
@@ -92,10 +97,32 @@ func (pt *safePortTracker) GetLastAssignedHostPort() int {
 var tracker safePortTracker
 
 // GetHostPortRange gets N contiguous host ports from the ephemeral host port range defined on the host.
+// dynamicHostPortRange can be set by customers using ECS Agent environment variable ECS_DYNAMIC_HOST_PORT_RANGE;
+// otherwise, ECS Agent will use the default value returned from GetDynamicHostPortRange() in the utils package.
 func GetHostPortRange(numberOfPorts int, protocol string, dynamicHostPortRange string) (string, error) {
 	portLock.Lock()
 	defer portLock.Unlock()
+	result, err := getHostPortsWithRange(numberOfPorts, protocol, dynamicHostPortRange)
+	return result, err
+}
+
+// GetHostPort gets 1 host port from the ephemeral host port range defined on the host.
+// dynamicHostPortRange can be set by customers using ECS Agent environment variable ECS_DYNAMIC_HOST_PORT_RANGE;
+// otherwise, ECS Agent will use the default value returned from GetDynamicHostPortRange() in the utils package.
+func GetHostPort(protocol string, dynamicHostPortRange string) (string, error) {
+	portLock.Lock()
+	defer portLock.Unlock()
+	numberOfPorts := 1
+	result, err := getHostPortsWithRange(numberOfPorts, protocol, dynamicHostPortRange)
+	if err == nil {
+		result = strings.Split(result, "-")[0]
+	}
+	return result, err
+}
 
+// getHostPortsWithRange returns the requested number of host ports using the given dynamic host port range
+// and protocol. If no host port(s) was/were found, an empty string along with the error message will be returned.
+func getHostPortsWithRange(numberOfPorts int, protocol, dynamicHostPortRange string) (string, error) {
```

Comment:
nit: `getNHostPorts` or something similar might be better suited? since this serves both singular port search and range search.

Reply:
Will rename it to `getNumOfHostPorts`. Thanks!